From a148b8623b7d97c87857f9e203bad538ebd837f0 Mon Sep 17 00:00:00 2001
From: Udit Kumar <udit.kumar@nxp.com>
Date: Wed, 12 Jun 2019 10:42:02 +0530
Subject: [PATCH] ahci_qoriq: bug fix for ecc_addr

Original driver expect a register node with name "sata-ecc"
this node is of 64 bit wide.
In ACPI such nodes can be provided with QWordMemory, but
QWordMemory can not hold DescriptorName more than 4 characters.

Therefore this patch changes platform property retrival based
upon index instead of named.

Signed-off-by: Udit Kumar <udit.kumar@nxp.com>
---
 drivers/ata/ahci_qoriq.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/ata/ahci_qoriq.c
+++ b/drivers/ata/ahci_qoriq.c
@@ -433,9 +433,7 @@ static int ahci_qoriq_probe(struct platf
 		qoriq_priv->type = (enum ahci_qoriq_type)acpi_id->driver_data;
 
 	if (unlikely(!ecc_initialized)) {
-		res = platform_get_resource_byname(pdev,
-						   IORESOURCE_MEM,
-						   "sata-ecc");
+		res = platform_get_resource(pdev, IORESOURCE_MEM, 1);
 		if (res) {
 			qoriq_priv->ecc_addr =
 				devm_ioremap_resource(dev, res);
